src/peft/tuners/tuners_utils.py

```diff
@@ -498,11 +514,18 @@ def inject_adapter(
             if len(new_target_modules) < len(peft_config.target_modules):
                 peft_config.target_modules = new_target_modules
 
+        ###############################
+        # MATCHING & CREATING MODULES #
+        ###############################
+
         existing_adapter_map = {}
         for key, module in named_modules:
             if isinstance(module, BaseTunerLayer):
                 existing_adapter_map[key] = module
 
+        # TODO: check if this the most robust way
+        state_dict_keys = {k.rsplit(".", 2)[0] for k in state_dict} if state_dict is not None else set()
```

Comment:
Are these state dict keys or module keys? I think it's the latter no? For example, `foo.bar.weight`. Here the module and submodule are `foo` and `bar`. If we were to just obtain state dict keys, a simple `state_dict.keys()` would have sufficed. So, I think we should consider renaming it.